controllers/factory/selectors.go

```diff
@@ -65,13 +66,7 @@ func visitObjectsWithSelector(ctx context.Context, rclient client.Client, ns []s
 		return nil
 	}
 
-	watchNamespace := config.MustGetWatchNamespace()
```

Comment:
It seems to me that MustGetWatchNamespace doesn't work the way you expect. ListOptions doesn't filter the list of namespaces by namespace - as a result all namespaces will always be in the list.